File: Merge Two Sorted Lists/code.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        if (l1 == null)
            return l2;
        if (l2 == null)
            return l1;

        ListNode newHead = null, originalHead = null;


        if (l1.val > l2.val)
        {
            newHead = l2;
            l2 = l2.next;
        }
        else
        {
            newHead = l1;
            l1 = l1.next;
        }

        originalHead = newHead;

        while ((l1 != null) && (l2 != null))
        {
            if (l1.val > l2.val)
            {
                newHead.next = l2;
                l2 = l2.next;
            }
            else
            {
                newHead.next = l1;
                l1 = l1.next;
            }

            newHead = newHead.next;
        }

        if (l1 == null)
        {
            newHead.next = l2;
        }
        else
        {
            newHead.next = l1;
        }

        return originalHead;
    }
}

File: Remove Nth Node From End of List/code.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode forwardHead = null, tempHead = head, originalHead = head;


            while (n > 0 && (tempHead.next != null)) {
                tempHead = tempHead.next;
                n--;
            }
            if (n == 0)
            {
                forwardHead = tempHead;
            }
            else {
                originalHead = originalHead.next;
                return originalHead;
            }

            while (forwardHead.next != null) {
                head = head.next;
                forwardHead = forwardHead.next;
            }
            if (head.next.next != null)
                head.next = head.next.next;
            else
                head.next = null;
            return originalHead;
    }
}